pkg/controller/ttl/ttlcontroller.go

```diff
@@ -0,0 +1,299 @@
+		{nodeCount: 2000, ttlSeconds: 60},
+		{nodeCount: 5000, ttlSeconds: 300},
+	}
+)
+
+func (ttlc *TTLController) getDesiredTTLSeconds(currentTTL int) int {
+	nodeCount := atomic.LoadInt32(&ttlc.nodeCount)
+	// To avoid situation when we are changing the annotation all the time,
+	// because the cluster size is flapping between X and X+1 nodes (e.g.
+	// due to cluster autoscaler), we use the following algorithm:
+	// - we compute the desired ttl first based on current cluster size.
+	// - if we are supposed to lower the current ttl, the current cluster
+	//   size must be at most boundary -1
+	// Note that with this algorith, to have flapping ttls on nodes, the cluster
+	// has to be flapping constantly betwenn <=X-1 and X+1<=.
+	// WARNING: you can't make this "1" a higher constant, as it may result
+	// in different nodes having different ttls over long period of time, e.g.
+	// if we swap the 1 with say 5, then in the scenario:
+	// - the cluster was of size 1500 => all nodes have ttl=60
+	// - cluster resized to 996 => all nodes still have ttl=60
+	// - one new node added => it will have ttl=30
+	// And nothing would change over time.
+	step := len(ttlBoundaries) - 1
+	for ; step > 0 && nodeCount <= ttlBoundaries[step-1].nodeCount; step-- {
+	}
+	desiredTTL := ttlBoundaries[step].ttlSeconds
+	for {
+		if step == len(ttlBoundaries)-1 {
+			break
+		}
+		if currentTTL <= desiredTTL || nodeCount <= ttlBoundaries[step].nodeCount-1 {
+			break
+		}
+		step++
+		desiredTTL = ttlBoundaries[step].ttlSeconds
+	}
+	return desiredTTL
+}
+
+func getIntFromAnnotation(node *v1.Node, annotationKey string) (int, bool) {
+	if node.Annotations == nil {
+		return 0, false
+	}
+	annotationValue, ok := node.Annotations[annotationKey]
+	if !ok {
+		return 0, false
+	}
+	intValue, err := strconv.Atoi(annotationValue)
+	if err != nil {
+		glog.Warningf("Cannot convert the value %q with annotation key %q for the node %q",
+			annotationValue, annotationKey, node.Name)
+		return 0, false
+	}
+	return intValue, true
+}
+
+func setIntAnnotation(node *v1.Node, annotationKey string, value int) {
+	if node.Annotations == nil {
+		node.Annotations = make(map[string]string)
+	}
+	node.Annotations[annotationKey] = strconv.Itoa(value)
+}
+
+func (ttlc *TTLController) patchNodeWithAnnotation(node *v1.Node, annotationKey string, value int) error {
+	oldData, err := json.Marshal(node)
+	if err != nil {
+		return err
+	}
+	setIntAnnotation(node, annotationKey, value)
+	newData, err := json.Marshal(node)
+	if err != nil {
+		return err
+	}
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, &v1.Node{})
+	if err != nil {
+		return err
+	}
+	_, err = ttlc.kubeClient.Core().Nodes().Patch(node.Name, types.StrategicMergePatchType, patchBytes)
+	if err != nil {
+		glog.V(2).Infof("Failed to change ttl annotation for node %s: %v", node.Name, err)
+		return err
+	}
+	glog.V(2).Infof("Changed ttl annotation for node %s: %v", node.Name, err)
```

Comment:
No need to print `err` since it's nil. Printing the actual TTL would be more useful.